lib/plugins/print/print.js

```diff
@@ -23,29 +25,76 @@ class Print {
     };
   }
 
+  adorn(svc) {
+    const service = svc;
+    // service object treatment
+    this.cache.serviceService = service.service;
+    if (_.isObject(this.cache.serviceService)) {
+      service.service = service.service.name;
+      service.serviceObject = this.cache.serviceService;
+    }
+    // provider treatment and defaults
+    this.cache.serviceProvider = service.provider;
+    if (_.isString(this.cache.serviceProvider)) {
+      service.provider = { name: this.cache.serviceProvider };
+    }
+    service.provider = _.merge({
```

Comment:
This should be only done if the provider is set to `aws`, shouldn't it? The print command is located at top level (outside of the aws plugin) and might be called for arbitrary provider types.

Reply:
I would have expected so too.  However, the code in ~/lib/classes/Service.js (https://github.com/serverless/serverless/blob/bf5a8c9fd4a90946f1862e1837e6899b542d4cb2/lib/classes/Service.js#L19) does this, I believe generally, for all services.  Do I misunderstand?  Otherwise, I'm happy to improve this.

Reply:
Indeed 🤔 . So it seems to be just an initial value there (that will be overridden by the incoming service configuration). In theory, a service.yml without provider definition would work then, which is a bit strange, but so be it.

However, here in the print command I'm not really sure if that, what we do here, is really just a preliminary intialization that will be overwritten or will influence subsequent behavior.

Reply:
Yeah, it was weird to see at first but I faulted on the side of parity to avoid bugs resulting from dissonance.  Thank you very much for paying attention here!